src/server/database/Database/Transaction.cpp

```diff
@@ -63,20 +67,51 @@ void Transaction::Cleanup()
     _cleanedUp = true;
 }
 
+const char* Transaction::ToString()
+{
+    std::string msg;
+    for (SQLElementData const& data : m_queries)
+    {
+        switch (data.type)
+        {
+        case SQL_ELEMENT_PREPARED:
+            msg += data.element.stmt->getQueryString() + ';';
+            break;
+        case SQL_ELEMENT_RAW:
+            msg += data.element.query + ';';
+            break;
+        }
+
+    }
+    return msg.c_str();
```

Comment:
Return the std::string instead to avoid crashes. Use c_str when you print the value only.